### PR TITLE
Add passthrough handler

### PR DESCRIPTION
**Pull request by jsha**
This passes through GET requests other than get-entries to the CT backend. This is particularly useful for some log scanning tools that call `/ct/v1/get-sth` before beginning their scan.

This is mainly intended as a convenience for small scale testing. In production we plan to bypass this tool for all request paths other than /ct/v1/get-entries.

### CODE REVIEW

**Review by aarongable — APPROVED** [review]
LGTM. Once this lands, https://github.com/letsencrypt/ctile/pull/17 will need to be updated to pass the `mux` instead of the `handler` on line 464.

File: main.go [hunk 1]
```diff
@@ -382,6 +375,35 @@ func singleflightDo[V any](group *singleflight.Group, key string, fn func() (V,
...
+}
+
+func (p passthroughHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		fmt.Fprintln(w, "only GET is supported")
+		return
+	}
+	url := fmt.Sprintf("%s%s", p.logURL, r.URL.Path)
+	r, err := http.NewRequestWithContext(r.Context(), http.MethodGet, url, nil)
```
Comment (beautifulentropy) [inline main.go]: You are reusing the input `r *http.Request` here.

```suggestion
	req, err := http.NewRequestWithContext(r.Context(), http.MethodGet, url, nil)
```

File: main.go [hunk 2]
```diff
@@ -382,6 +375,35 @@ func singleflightDo[V any](group *singleflight.Group, key string, fn func() (V,
...
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "creating request: %s\n", err)
+		return
+	}
+	resp, err := http.DefaultClient.Do(r)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "fetching %s: %s\n", url, err)
+		return
+	}
```
Comment (beautifulentropy) [inline main.go:401]: I think `resp.Body` needs to be closed before we return.

```suggestion
	}
	defer resp.Body.Close()
```

**Review by beautifulentropy — CHANGES_REQUESTED** [review]

File: main.go [hunk 3]
```diff
@@ -382,6 +375,35 @@ func singleflightDo[V any](group *singleflight.Group, key string, fn func() (V,
...
+
+	w.WriteHeader(resp.StatusCode)
+	io.Copy(w, resp.Body)
```
Comment (beautifulentropy) [inline main.go]: `io.Copy` could return an error and it would be going unchecked.

**Review by beautifulentropy — APPROVED** [review]

**Review by aarongable — APPROVED** [review]

**Pull request merged**